test/test_images/probe_helper/probe_helper.go

```diff
@@ -87,51 +106,141 @@ func forwardFromProbe(ctx context.Context, brokerClient cloudevents.Client, pubs
 		defer cancel()
 		switch event.Type() {
 		case BrokerE2EDeliveryProbeEventType:
-			receiverChannel, err = receivedEvents.createReceiverChannel(event)
+			channelID = event.ID()
+			receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
 			if err != nil {
-				return cloudevents.NewReceipt(false, "Probe forwarding failed, could not create receiver channel: %v", err)
+				log.Printf("Probe forwarding failed, could not create receiver channel: %v", err)
+				return cloudevents.ResultNACK
 			}
+			defer receivedEvents.deleteReceiverChannel(channelID)
+
 			// The broker client forwards the event to the broker.
 			if res := brokerClient.Send(ctx, event); !cloudevents.IsACK(res) {
 				log.Printf("Error when sending event %v to broker: %+v \n", event.ID(), res)
 				return res
 			}
 		case CloudPubSubSourceProbeEventType:
-			receiverChannel, err = receivedEvents.createReceiverChannel(event)
+			channelID = event.ID()
+			receiverChannel, err = receivedEvents.createReceiverChannel(channelID)
 			if err != nil {
-				return cloudevents.NewReceipt(false, "Probe forwarding failed, could not create receiver channel: %v", err)
+				log.Printf("Probe forwarding failed, could not create receiver channel: %v", err)
+				return cloudevents.ResultNACK
 			}
+			defer receivedEvents.deleteReceiverChannel(channelID)
+
 			// The pubsub client forwards the event as a message to a pubsub topic.
 			if res := pubsubClient.Send(ctx, event); !cloudevents.IsACK(res) {
 				log.Printf("Error when publishing event %v to pubsub topic: %+v \n", event.ID(), res)
 				return res
 			}
+		case CloudStorageSourceProbeEventType:
+			obj := bucket.Object(event.ID())
+			switch event.Subject() {
+			case CloudStorageSourceProbeCreateSubject:
```

Comment:
Can we use the `schemasv1.CloudStorageObjectFinalizedEventType` as event type and get rid of the subject?

Reply:
This is doable but requires some changes. Currently, the object created by the probe is named by the event ID, which is prefixed by the event type (which is the probe type), e.g. event.Type() == cloudstoragesource-probe, event.ID() == cloudstoragesource-probe-1234567890. If we want the distinct probes here to be able to reference the same object without having the same event type, then we can either change the event ID scheme to not contain the event type (not a bad idea in my opinion), or add some extra string processing in the probe helper to remove the event type prefix from the event ID (not so elegant). Thoughts?

Reply:
Thought about this a bit more. The "Event Type" in the probe request is more like "Probe Type", so it doesn't necessarily map to the real event type.  I think what you have is OK.